uploader/pantheon.py

```diff
@@ -0,0 +1,214 @@
+        return config[configKey]
+    else:
+        return default
+
+
+server = resolveOption(args.server, 'server', DEFAULT_SERVER)
+repository = resolveOption(args.repository, 'repository', DEFAULT_REPOSITORY)
+links = resolveOption(args.links, 'followlinks', DEFAULT_LINKS)
+
+print('Using server: ' + server)
+print('Using repository: ' + repository)
+print('--------------')
+
+titleGlobs = config['titles'] if config is not None else ()
+moduleGlobs = config['modules'] if config is not None else ()
+resourceGlobs = config['resources'] if config is not None else '*'
+unspecified_files = 0
+logger.debug('titleGlobs: %s', titleGlobs)
+logger.debug('moduleGlobs: %s', moduleGlobs)
+logger.debug('resourceGlobs: %s', resourceGlobs)
+
+
+for root, dirs, files in os.walk(args.directory, followlinks=links):
+    for file in files:
+        if file == 'pantheon2.yml':
+            continue
+        logger.debug('root: %s', root)
+        logger.debug('file: %s', file)
+        path = PurePath(root + '/' + file)
+
+        # These distinctions aren't important right now but they set us up for later
+        isTitle = matches(path, titleGlobs, 'titles')
+        isModule = matches(path, moduleGlobs, 'modules') if not isTitle else False
+        isResource = matches(path, resourceGlobs, 'resources') if not isModule else False
+        url = server + "/content/repositories/" + repository
+        #logger.debug('isTitle: %s', isTitle)
+        #logger.debug('isModule: %s', isModule)
+        #logger.debug('isResource: %s', isResource)
+        if isModule or isTitle or isResource:
+            base_name = path.stem
+
+            ppath = path
+            hiddenFolder = False
+            while not ppath == PurePath(args.directory):
+                logger.debug('ppath: %s', str(ppath.stem))
+                if ppath.stem[0] == '.':
+                    hiddenFolder = True
+                    break
+                ppath = ppath.parent
+            if hiddenFolder:
+                logger.info('Skipping %s because it is hidden.', str(path))
+                logger.info('')
+                continue
+
+            # parent directory
+            parent_dir_str = str(path.parent.relative_to(args.directory))
+            if parent_dir_str == '.':
+                parent_dir_str = ''
+            logger.debug('parent_dir_str: %s', parent_dir_str)
+            # file becomes a/file/name (no extension)
+
+
+            if parent_dir_str:
+                url += '/' + parent_dir_str
+
+            logger.debug('base name: %s', base_name)
+
+            # Asciidoc content (treat as a module)
+            if path.suffix == '.adoc' or path.suffix == '.asciidoc':
+                print(path)
+                url += '/' + path.name
+                logger.debug('url: %s', url)
+                jcr_primary_type = "pant:module" if isModule else "pant:title"
+                sling_resource_type = "pantheon/modules" if isModule else "pantheon/titles"
```

Comment:
Same comment as before here - can we remove `sling_resource_type` here? This should get autocreated for us via `nodetypes.cnd`.

Reply:
done